### PR TITLE
[otbn] Specify life cycle escalation reaction

### LINKED ISSUE #4227: [otbn] Specify what a LC_ESCALATE_EN does
OTBN should wipe internal state when it receives the LC_ESCALATE_EN signals. We need to specify what exactly is meant by that (and later implement it).

> estimate 2

### PR DESCRIPTION
**Pull request by imphil**
Fixes #4227

### CODE REVIEW

**Comment by imphil** [comment]
This PR works much better with https://github.com/lowRISC/opentitan/pull/7433, but doesn't necessarily depend on it; since I wouldn't be surprised if we find things to discuss in #7433 I wanted to split this out as a minimum solution to our sprint goal.

**Review by msfschaffner — APPROVED** [review]
LGTM

File: hw/ip/otbn/doc/_index.md [hunk 1]
```diff
@@ -565,6 +564,12 @@ The error that caused the alert can be determined by reading the {{< regref "FAT
 If OTBN was running, this value will also be reflected in the {{< regref "ERR_BITS" >}} register.
 A fatal alert can only be cleared by resetting OTBN through the `rst_ni` line.
 
+### Reaction to Life Cycle Escalation Requests
+
+OTBN receives and reacts to escalation signals from the [life cycle controller]({{< relref "/hw/ip/lc_ctrl/doc#security-escalation" >}}).
+
+An escalation request signaled through the `lc_escalate_en_i` signal results in the same action as a [fatal error](#design-details-fatal-errors) without raising a fatal alert.
```
Comment (rswarbrick) [inline hw/ip/otbn/doc/_index.md:571]: Maybe "... but does not raise a fatal alert"? I think it's a bit clearer that way?

**Review by rswarbrick — APPROVED** [review]
Looks good!

**Pull request merged**